### ISSUE TITLE
[Bug]: JSON Form Mapped from Table does not maintain the order of Columns onto its Fields

### ISSUE THREAD

**Issue by Aishwarya-U-R** [appsmithorg/appsmith#13666]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

**Table columns order not maintained & Date format applied in Table column not honoured:**

<img width="1321" alt="Screenshot 2022-05-09 at 12 26 39 PM" src="https://user-images.githubusercontent.com/91450662/167356975-32ca22bd-1587-4dec-a80a-ac43f3684319.png">

**Table hidden fields not honoured:**
<img width="1029" alt="Screenshot 2022-05-09 at 12 32 19 PM" src="https://user-images.githubusercontent.com/91450662/167357457-acc0ec69-525a-47cd-add4-f5176f296b02.png">

### Steps To Reproduce

1. Widgets needed: Table, JSON Widget
2. Map some data onto Table widget
3. Connect to JSON form for view
4. **Observe : Order of Table column not maintained in JSON form fields**
5. **Observe: Data format applied is not carry forwarded**
6. **Observe: Hidden columns also shown in JSON form field**
7. **Expected: All above should work :)** 

### Version

Prod/Release v1.6.23

**Comment by dilippitchika** [comment]
Fix should be to re-order the table's selectedRow and triggeredRow property keys to be the same as the order set by -

1. Developer - This is straightforward, whatever the order is set by developer it should be the output for selectedRow and triggeredRow
2. User - I don't know if we should do this but this means the form changes order when user re-orders anything in the table. Need suggestions, in my opinion currently we should not do this.

Regarding other issues
- Observe: Date format applied does not reflect
- Observe: Column name change from Table does not reflect in JSON form
- Observe: Hidden columns also shown in JSON form field

This means a tighter coupling between json form and a table, to auto-infer these changes from the table. This is a new feature request, as table and json form are 2 independent widgets and cannot work well without manual intervention. 

The same can be said for any scenario where table1.selectedRow.date is being used in a datepicker widget directly that the format doesn't match, so i believe this is a not just for a table widget and json form, but any widget in appsmith.

**Comment by dilippitchika** [comment]
## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 200 |
| Effort (months)  | 0.5 |

**Comment by Nikhil-Nandagopal** [comment]
Agreed, I think we should avoid this. It can cause some unexpected behavior.
